src/main/kotlin/fi/metatavu/oss/api/persistence/repository/AbstractRepository.kt

```diff
@@ -0,0 +1,63 @@
+package fi.metatavu.oss.api.persistence.repository
+
+import io.quarkus.hibernate.reactive.panache.PanacheQuery
+import io.quarkus.hibernate.reactive.panache.PanacheRepositoryBase
+import io.smallrye.mutiny.Uni
+
+abstract class AbstractRepository<Entity, Id> : PanacheRepositoryBase<Entity, Id> {
+
+    /**
+     * Lists all entities
+     *
+     * @param page page
+     * @param pageSize page size
+     * @return entities
+     */
+    open fun listAllWithPaging(page: Int? = null, pageSize: Int? = null): Pair<Uni<List<Entity>>, Uni<Long>> {
```

Comment:
Should the unis be awaited here and return the actual entity? Maybe this could be consulted from Antti. Same thing applies to all methods in this class.

Reply:
Yes, unis are required for @reactivetransaction annotations to work since we removed the transactions from the db access level. It'll be visible in the next Surveys pr.